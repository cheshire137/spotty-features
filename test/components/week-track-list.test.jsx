import React from 'react'
import renderer from 'react-test-renderer'

import SavedTrackResponse from '../fixtures/spotify/saved-tracks'

import WeekTrackList from '../../src/components/week-track-list.jsx'

const weekStr = '2017-02-12T06:00:00.000Z'
const item = SavedTrackResponse.items[0]
const track = {
  id: item.track.id,
  savedAt: new Date(item.added_at),
  week: new Date(weekStr),
  name: item.track.name,
  artists: item.track.artists.map(artist => artist.name),
  album: item.track.album.name,
  url: item.track.external_urls.spotify,
  albumUrl: item.track.album.external_urls.spotify,
  audioFeatures: {
    acousticness: 0.145,
    danceability: 0.468,
    energy: 0.515,
    instrumentalness: 0,
    liveness: 0.132,
    loudness: -6.578,
    speechiness: 0.0665,
    valence: 0.186,
    negativity: 0.814
  }
}

function props() {
  return {
    week: weekStr,
    tracks: [track],
    avgLoudness: track.audioFeatures.loudness
  }
}

describe('WeekTrackList', () => {
  let component = null

  beforeEach(() => {
    component = <WeekTrackList {...props()} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
