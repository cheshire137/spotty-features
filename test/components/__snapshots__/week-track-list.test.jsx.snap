exports[`WeekTrackList matches snapshot 1`] = `
<dl
  className="week-track-list">
  <dt
    className="title is-3">
    Week of February 12, 2017
    <span
      className="track-count">
      Saved 1 track
    </span>
  </dt>
  <dd>
    <ul>
      <li>
        <div
          className="columns">
          <div
            className="column">
            <a
              className="track-link"
              href="https://open.spotify.com/track/4nDfc6F2uVcc6wdG7kBzWO"
              rel="noopener noreferrer"
              target="_blank">
              <span
                className="track-name">
                Bom Bidi Bom
              </span>
              <span>
                 by 
              </span>
              <span
                className="track-artists">
                Nick Jonas, Nicki Minaj
              </span>
            </a>
            <a
              className="track-link"
              href="https://open.spotify.com/album/4RPo8gaqAbJksSE8JOIVLF"
              rel="noopener noreferrer"
              target="_blank">
              <span
                className="track-album">
                Fifty Shades Darker (Original Motion Picture Soundtrack)
              </span>
            </a>
            <span
              className="track-saved-at">
              Saved 
              2/17/2017
            </span>
            <p
              className="audio-features">
              <span
                style={
                  Object {
                    "color": "#e3b951",
                  }
                }
                title="52%">
                Energetic
              </span>
              <span
                className="loudness-quiet">
                -6.578
                 dB
              </span>
              <span
                style={
                  Object {
                    "color": "#004DD0",
                  }
                }
                title="19% positive">
                Negative
              </span>
            </p>
          </div>
        </div>
      </li>
    </ul>
  </dd>
</dl>
`;
